testing/libc++/APKBUILD

```diff
@@ -0,0 +1,94 @@
+# Contributor: Dmitry Golovin <dima@golovin.in>
+# Contributor: Shiz <hi@shiz.me>
+# Maintainer: Dmitry Golovin <dima@golovin.in>
+pkgname=libc++
+pkgver=4.0.0
+pkgrel=0
+_llvmver=${pkgver%%.*}
+pkgdesc="A new implementation of the C++ standard library, targeting C++11"
+url="http://libcxx.llvm.org/"
+arch="all"
+license="UOI-NCSA"
+depends_dev="${pkgname}abi-dev=$pkgver-r$pkgrel"
+makedepends="cmake llvm-dev>=$_llvmver llvm-static>=$_llvmver clang>=$_llvmver"
+checkdepends="lit"
+subpackages="$pkgname-dev ${pkgname}abi:abi ${pkgname}abi-dev:abi_dev"
+source="http://releases.llvm.org/$pkgver/libcxx-$pkgver.src.tar.xz
+	http://releases.llvm.org/$pkgver/libcxxabi-$pkgver.src.tar.xz
+	http://releases.llvm.org/$pkgver/libunwind-$pkgver.src.tar.xz
+        check-for-musl-libcs-max_align_t.patch"
+builddir="$srcdir/libcxx-$pkgver.src"
+_abidir="$srcdir/libcxxabi-$pkgver.src"
+_unwinddir="$srcdir/libunwind-$pkgver.src"
+
+build() {
+	mkdir -p "$_abidir"/build
+	cd "$_abidir"/build
+        cmake .. \
+                -DCMAKE_BUILD_TYPE=MinSizeRel \
+                -DCMAKE_INSTALL_PREFIX=/usr \
+		-DCMAKE_CXX_COMPILER=clang++ \
+                -DCMAKE_CXX_FLAGS="$CXXFLAGS" \
+		-DCMAKE_C_COMPILER=clang \
+                -DCMAKE_C_FLAGS="$CFLAGS" \
+                -DLIBCXXABI_USE_LLVM_UNWINDER=ON \
+                -DLIBCXXABI_LIBUNWIND_PATH="$_unwinddir" \
+                -DLIBCXXABI_LIBCXX_INCLUDES="$builddir"/include \
+                -DLIBCXXABI_INCLUDE_TESTS=1
+	make
+
+	mkdir -p "$builddir"/build
+	cd "$builddir"/build
+	cmake .. \
+		-DCMAKE_BUILD_TYPE=MinSizeRel \
+		-DCMAKE_INSTALL_PREFIX=/usr \
+		-DCMAKE_CXX_COMPILER=clang++ \
+		-DCMAKE_CXX_FLAGS="$CXXFLAGS" \
+		-DCMAKE_C_COMPILER=clang \
+		-DCMAKE_C_FLAGS="$CFLAGS" \
+		-DCMAKE_VERBOSE_MAKEFILE=ON \
+		-DLIBCXX_HAS_MUSL_LIBC=ON \
+		-DLIBCXX_HAS_GCC_S_LIB=OFF \
+		-DLIBCXX_CXX_ABI=libcxxabi \
+		-DLIBCXX_CXX_ABI_INCLUDE_PATHS="$_abidir"/include \
+		-DLIBCXX_CXX_ABI_LIBRARY_PATH="$_abidir"/build/lib \
+		-DLIBCXX_ENABLE_STATIC_ABI_LIBRARY=ON  # needed to make -static work
+	make
+}
+
+check() {
+	cd "$_abidir"/build
+	# some of these tests fail for now
+	make check-cxxabi || true
+
+	cd "$builddir"/build
+	# this testsuite takes forever, and some tests still fail
+	#make check-cxx || true
+}
+
+package() {
+	depends="${pkgname}abi>=$_llvmver"
+
+	cd "$builddir"/build
+	make install DESTDIR="$pkgdir"
+}
+
+abi() {
+	pkgdesc="A new implementation of low level support for a standard C++ library"
+	depends="llvm-libunwind>=$_llvmver"
+
+	cd "$_abidir"/build
+	make install DESTDIR="$subpkgdir"
+}
+
+abi_dev() {
+	local oldpkgdir; oldpkgdir="$pkgdir"
```

Comment:
Maybe some ancient shell from prehistory? :) Really, I don’t know any commonly used shell that does not support this. And we already use it everywhere.